presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -574,17 +561,20 @@ ObjectMetadata getS3ObjectMetadata(Path path)
                                         throw new UnrecoverableS3OperationException(path, e);
                                 }
                             }
-                            throw Throwables.propagate(e);
+                            throw e;
                         }
                     });
         }
         catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new RuntimeException(e);
         }
+        catch (IOException e) {
+            throw e;
+        }
         catch (Exception e) {
-            throwIfInstanceOf(e, IOException.class);
-            throw Throwables.propagate(e);
+            throwIfUnchecked(e);
+            throw new RuntimeException(e);
```

Comment:
The code used to rethrow `IOException` as-is. Is it intentional that it no longer does so?

(here and i noticed the same in 2 other places)

Reply:
There is a `catch` above to catch and rethrow `IOException`s.

Reply:
I removed it anyway as it wasn't really a necessary change.